src/lib/transition_frontier/transition_frontier.ml

```diff
@@ -331,8 +345,39 @@ module Make (Inputs : Inputs_intf) :
     Coda_metrics.(Gauge.set Transition_frontier.active_breadcrumbs 1.0) ;
     t
 
-  let close {extension_writers; _} =
+  let incr_num_catchup_jobs
+      {num_catchup_jobs= {mvar; signal_writer; _}; logger; _} =
+    let%bind current_num_catchup_jobs = Mvar.take mvar in
+    Logger.info logger "Incrementing num catch up jobs. Current number %i"
+      current_num_catchup_jobs ~module_:__MODULE__ ~location:__LOC__ ;
+    let%bind () =
+      if current_num_catchup_jobs = 0 then
+        Broadcast_pipe.Writer.write signal_writer `Catchup
+      else Deferred.unit
+    in
+    Mvar.put mvar (current_num_catchup_jobs + 1)
+
+  let decr_num_catchup_jobs
+      {num_catchup_jobs= {mvar; signal_writer; _}; logger; _} =
+    let%bind current_num_catchup_jobs = Mvar.take mvar in
+    Logger.info logger "Decrementing num catch up jobs. Current number % i"
```

Comment:
How noisy are these logs? If too noisy please move to debug or trace. If not too noisy then we keep

Reply:
I imagine these logs will be quite noisy when the slot window is small, but with long slot windows, this should probably be rare (as in, only when you first finish bootstrapping).

Reply:
whoops. Suppose to be in trace.